tuf/ngclient/fetcher.py

```diff
@@ -40,3 +48,51 @@ def fetch(self, url: str, required_length: int) -> Iterator[bytes]:
             A bytes iterator
         """
         raise NotImplementedError  # pragma: no cover
+
+    @contextmanager
+    def download_file(self, url: str, required_length: int) -> Iterator[IO]:
+        """Opens a connection to 'url' and downloads the content
+        up to 'required_length'.
+
+        Args:
+          url: a URL string that represents the location of the file.
+          required_length: an integer value representing the length of
+              the file or an upper boundary.
+
+        Raises:
+          DownloadLengthMismatchError: a mismatch of observed vs expected
+              lengths while downloading the file.
+
+        Yields:
+          A file object that points to the contents of 'url'.
+        """
+        # 'url.replace('\\', '/')' is needed for compatibility with
+        # Windows-based systems, because they might use back-slashes in place
+        # of forward-slashes. This converts it to the common format.
+        # unquote() replaces %xx escapes in a url with their single-character
+        # equivalent. A back-slash may beencoded as %5c in the url, which
+        # should also be replaced with a forward slash.
+        url = parse.unquote(url).replace("\\", "/")
```

Comment:
let's get rid of this: this component should expect to get a real URL and should not do this.

Do we need a new issue to handle possible issues in higher level code?

Reply:
Opened a new issue although there are several ones related to paths and urls in Updater already: https://github.com/theupdateframework/tuf/issues/1483

Reply:
oh you might be right about multiple bugs -- I probably file one every time I see this code